### PR TITLE
Make rustpython-vm compatible with non-js wasm32-unknown & add tests

### PR DESCRIPTION
**Pull request by coolreader18**
Ideally the test crate shouldn't take long to compile, since it's the same target with only a few crates configured differently.

### CODE REVIEW

**Comment by coolreader18** [comment]
Hmm, is there a reason we need to have wasm32-unknown-unknown checked in exotic_targets when we have the wasm job? Is that a holdover from before we had the wasm job?

**Comment by youknowone** [comment]
I recently added it there without specific reason. Do you think it belongs to wasm job? That makes sense.

File: wasm/wasm-unknown-test/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+A test crate to ensure that `rustpython-vm` compiles on `wasm32-unknown-unknown` without a JS host.
```
Comment (youknowone) [inline wasm/wasm-unknown-test/README.md:1]: :+1:

**Comment by youknowone** [comment]
I added the build to prevent the target's build error.

wasm build job was not fully covering the top-most crate rustpython build
See #3634

**Comment by coolreader18** [comment]
Oh, I see. But that's not supported anyway, right? You can't compile a cli application to wasm w/ js-interop, there's no syscalls or argv or anything.

**Comment by youknowone** [comment]
there was wasm32-unknown use case without js interop.
the top crate is now more useful than vm because user must manually construct interpreter using vm and stdlib since we seprated stdlib crate. `rustpython` crate suggests a convenient way to construct one without knowing the details.

**Comment by youknowone** [comment]
cc @lastmjs

**Review by youknowone — APPROVED** [review]
not sure if skipping `cargo check --target wasm32-unknown-unknown --no-default-features` is safe.
other parts looks great

File: vm/src/stdlib/time.rs [hunk 2]
```diff
@@ -102,6 +102,11 @@ mod time {
         Ok(Date::now() / 1000.0)
     }
 
+    #[cfg(all(target_arch = "wasm32", not(feature = "js"), not(target_os = "wasi")))]
+    fn _time(vm: &VirtualMachine) -> PyResult<f64> {
+        Err(vm.new_not_implemented_error("time.time".to_owned()))
```
Comment (lastmjs) [inline vm/src/stdlib/time.rs:107]: I will need a way to implement time, I have the APIs in my environment, but I will need a way to override this

File: vm/src/vm/vm_object.rs [hunk 3]
```diff
@@ -53,6 +53,11 @@ impl VirtualMachine {
             error(&s);
             panic!("{}; exception backtrace above", msg)
         }
+        #[cfg(all(target_arch = "wasm32", not(feature = "js"), not(target_os = "wasi")))]
```
Comment (lastmjs) [inline vm/src/vm/vm_object.rs:56]: Wouldn't it be useful to still see information from the exception? Can't you do something like:

```rust
let err_string: String = exc.to_pyobject(vm).repr(vm).unwrap().to_string();
```

Or does the `msg` have everything we need?

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: vm/src/vm/vm_object.rs (hunk 3 above)
Comment (youknowone) [reply vm/src/vm/vm_object.rs:56]: It will be useful

**Pull request closed without merging**